common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -228,6 +228,8 @@ def load_item(self, location):
         Return an XModule instance for the specified location
         """
         assert isinstance(location, UsageKey)
+        if location.run is None:
+            location = location.replace(course_key=self.modulestore.fill_in_run(location.course_key))
```

Comment:
If I understand this performance problem correctly, the lack of a run was causing a cache miss. A comment as to why the run is being filled in here would be splendid.